docs/shell-tips.md

```diff
@@ -164,10 +194,134 @@ have the newline character, and thus the `\<newline>`.
+  have to add an additional argument after it (`/home/john/backups` in the
+  example).
+
+- `--mouse` is just an option you set, `nano` doesn't need anything else to
+  make it work. Thus, there isn't any extra argument for that.
+
+Note that the `todo.txt` is the file we want to open! It has nothing to do with
+the previous argument. This will probably clarify it (taken from `nano`'s
+help):
+
+```
+Usage: nano [OPTIONS] [FILE]...
+```
+
+So, in the options you indicate the arguments, and `FILE` is... well, the file.
+
+Cool, huh?
+
+## Shebangs
+
+You can run some files directly, without specifying a program to interpret
+them.
+
+That's why you may have seen cases when some Python scripts are called with
+`python`:
+
+```
+$ python my_program.py
+```
+
+While other times, `python` isn't used:
+
+```
+$ ./my_program.py
+```
+
+In the latter, it's skipped because `my_program.py` already specifies in it
+which interpreter should be used (in this case, `python`).
+
+This is indicated in the very first line of the script files, and it's called
+a **shebang**. It looks like this, in the case of Python scripts:
+
+```
+#!/usr/bin/env python
+```
+
+With this, you're telling the shell: "if I tell you to run this, ask
+`/usr/bin/env python` how to understand it".
+
+`/usr/bin/env` is a way to identify where `python` is installed. If it was in
+`/usr/bin/python`, you could use the shebang `#!/usr/bin/python`, but `env`
+allows more flexibility (since not everyone has their Python interpreter
+there).
+
+Another frequent example of shebang is the one used in Bash scripts. In those
+cases, `#!/bin/sh` is used.
+
+The result is that the shell calls the program specified in the shebang, with
+the script as a parameter.  So, returning to our example with `my_program.py`,
+when you run `./my_program.py`, what happens under the hood is:
+
+```
+$ /usr/bin/env python my_program.py
+```
+
 ## Understanding commands
 
 Frequently, you may find commands that you don't understand, or don't
 know what they do. You can use `man <command>` to see the **man**ual page for
 that specific command. Also, you may find useful
 [explainshell](http://explainshell.com/), a webpage that explains what most
 commands do, part by part.
+
+## Cheatsheet
+
+There are many more commands in the shell, besides the ones explained in this
+file.
+[Here](https://www.git-tower.com/blog/command-line-cheat-sheet/) you can find
+a simple yet useful cheatsheet, created by Tower, that you may find useful
+to understand and remember what other common commands do (like `ls`, for
+instance).
+
+## Git
+
+Probably at this point you've heard about Git. It's basically a tool that most
+developers use in order to keep a control of all the changes made in their
+code.
+
+At first it seems witchcraft, but once you get the basic concepts you find it
+extremely useful and even easy to use (at least the 99% percent of the times).
```

Comment:
Whooops, missed that! BTW, I didn't know that was called a pleonasm, so thanks for the new word :)